be/src/storage/lake/update_manager.cpp

```diff
@@ -401,6 +398,55 @@ void UpdateManager::remove_primary_index_cache(uint32_t tablet_id) {
     LOG(WARNING) << "Lake update manager remove primary index cache, tablet_id: " << tablet_id << " , succ: " << succ;
 }
 
+bool UpdateManager::check_meta_version(const Tablet& tablet, int64_t base_version, int64_t new_version) {
+    auto index_entry = _index_cache.get(tablet.id());
+    if (index_entry == nullptr) {
+        // if primary index not in cache, check tablet meta directly
+        if (fs::path_exist(tablet.metadata_location(new_version))) {
+            return false;
+        } else {
+            return true;
+        }
+    } else {
+        auto& index = index_entry->value();
+        CheckVersionResult res = index.check_data_version(base_version);
+        if (res == CheckVersionResult::SUCCESS) {
+            // success
+            _index_cache.release(index_entry);
+            return true;
+        } else if (res == CheckVersionResult::ADVANCE) {
+            // advance, remove index and return success
+            _index_cache.remove(index_entry);
```

Comment:
Let `_index_cache.remove()` return true on success and check the return value to ensure the index really been removed.